### ISSUE TITLE
How to print image load order in ultralytics

### ISSUE THREAD

**Issue by slwgsc** [ultralytics/ultralytics#16863]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

How to print out the order in which the model loads the images of the dataset to verify that the model is trained and reasoned in the order given in the dataset, and can you give the specific modification code for this?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @slwgsc, thank you for your interest in Ultralytics 🚀! We recommend checking the [Docs](https://docs.ultralytics.com) for guidance, where you can find useful examples and methods related to data handling.

If this involves a specific 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to assist us in debugging.

If this is a ❓ Question about custom training or image processing, we suggest sharing more details about your setup or what you aim to achieve. This could include data samples or configurations related to image loading.

Feel free to join our community for discussions! Engage in real-time on [Discord](https://ultralytics.com/discord) 🎧, have in-depth chats on [Discourse](https://community.ultralytics.com), or connect with others on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Ensure that you are using the latest `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) met in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

You can also explore YOLO on various verified environments with preinstalled dependencies:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If the CI badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing, ensuring functionality across platforms. 

This is an automated response; an Ultralytics engineer will further assist you soon. 😊

**Comment by Y-T-G** [comment]
The training dataset is shuffled each epoch. So there's no particular order.

After you finish training, you can access the train dataloader at `model.trainer.train_loader`

```python
>>> for i in model.trainer.train_loader:
...     print(i["im_file"])
...     break                                                     ...
['/datasets/coco128/images/train2017/000000000536.jpg', '/datasets/coco128/images/train2017/000000000542.jpg', '/datasets/coco128/images/train2017/000000000071.jpg', '/datasets/coco128/images/train2017/000000000562.jpg']
```

It's going to be different each time.
